lib/vicadmin/validate.go

```diff
@@ -252,13 +255,25 @@ func (v *Validator) QueryDatastore(ctx context.Context, vch *config.VirtualConta
 		}
 	}
 
+	if sess.Client == nil {
+		return fmt.Errorf("Non-nil session govmomi client is nil while looking for datastore info")
+	}
+
+	if sess.Client.Client == nil {
+		return fmt.Errorf("Non-nil session vim client is nil while looking for datastore info")
+	}
+
 	pc := property.DefaultCollector(sess.Client.Client)
+	if pc == nil {
+		return fmt.Errorf("Could not get default propery collector; prop-collector came back nil")
+	}
+
 	err := pc.Retrieve(ctx, refs, nil, &dataStores)
```

Comment:
Well I had a `return err` there originally but @emlin told me not to return err there, because there may be other datastores that are working. I misremembered and thought that the properycollector could come back nil, not `refs`